pkg/linodeclient/stubclient/stubclient.go

```diff
@@ -0,0 +1,352 @@
+// This is a stub function.
+func New(opts ...Option) *Client {
+	c := &Client{
+		objectStorageBuckets:        make(map[string]*linodego.ObjectStorageBucket),
+		objectStorageBucketAccesses: make(map[string]*linodego.ObjectStorageBucketAccess),
+		objectStorageKeys:           make(map[int]*linodego.ObjectStorageKey),
+	}
+
+	for _, opt := range opts {
+		opt(c)
+	}
+
+	return c
+}
+
+func validateACL(acl linodego.ObjectStorageACL) bool {
+	switch acl {
+	case linodego.ACLPrivate, linodego.ACLAuthenticatedRead, linodego.ACLPublicRead, linodego.ACLPublicReadWrite:
+		return true
+	default:
+		return false
+	}
+}
+
+// CreateObjectStorageBucket is a stub function that stubs the behavior of CreateObjectStorageBucket call from linodego.Client.
+func (c *Client) CreateObjectStorageBucket(ctx context.Context, opt linodego.ObjectStorageBucketCreateOptions) (*linodego.ObjectStorageBucket, error) {
+	if v := ctx.Value(ForcedFailure); v != nil {
+		switch v := v.(type) {
+		case error:
+			return nil, v
+		default:
+			return nil, ErrUnexpectedError
+		}
+	}
+
+	key := fmt.Sprintf("%s/%s", opt.Cluster, opt.Label)
+
+	bucket, ok := c.objectStorageBuckets[key]
+	if ok {
+		return bucket, nil
+	}
+
+	if !validateACL(opt.ACL) {
+		return nil, &linodego.Error{
+			Code: http.StatusBadRequest,
+		}
+	}
+
+	bucket = &linodego.ObjectStorageBucket{
+		Label:    opt.Label,
+		Cluster:  opt.Cluster,
+		Hostname: fmt.Sprintf("%s.linodeobjects.com", opt.Label),
+	}
+	c.objectStorageBuckets[key] = bucket
+
+	corsEnabled := false
+	if opt.CorsEnabled != nil {
+		corsEnabled = *opt.CorsEnabled
+	}
+
+	access := &linodego.ObjectStorageBucketAccess{
+		ACL:         opt.ACL,
+		CorsEnabled: corsEnabled,
+	}
+	c.objectStorageBucketAccesses[key] = access
+
+	return bucket, nil
+}
+
+// GetObjectStorageBucket is a stub function that stubs the behavior of GetObjectStorageBucket call from linodego.Client.
+func (c *Client) GetObjectStorageBucket(ctx context.Context, clusterID, label string) (*linodego.ObjectStorageBucket, error) {
+	if v := ctx.Value(ForcedFailure); v != nil {
+		switch v := v.(type) {
+		case error:
+			return nil, v
+		default:
+			return nil, ErrUnexpectedError
+		}
+	}
+
+	key := fmt.Sprintf("%s/%s", clusterID, label)
+
+	obj, ok := c.objectStorageBuckets[key]
+	if ok {
+		return obj, nil
+	}
+
+	return nil, &linodego.Error{
+		Code: http.StatusNotFound,
+	}
+}
+
+// DeleteObjectStorageBucket is a stub function that stubs the behavior of DeleteObjectStorageBucket call from linodego.Client.
+func (c *Client) DeleteObjectStorageBucket(ctx context.Context, clusterID, label string) error {
+	if v := ctx.Value(ForcedFailure); v != nil {
+		switch v := v.(type) {
+		case error:
+			return v
+		default:
+			return ErrUnexpectedError
+		}
+	}
+
+	key := fmt.Sprintf("%s/%s", clusterID, label)
+
+	bucket, ok := c.objectStorageBuckets[key]
+	if ok && bucket.Objects == 0 && bucket.Size == 0 {
```

Comment:
suggestion: Turn the rest into a switch, personal opinion is I think it's easier to read

Reply:
```
switch {
	case ok && (bucket.Objects == 0 && bucket.Size == 0):
		delete(c.objectStorageBuckets, key)
		return nil
	case ok && (bucket.Objects != 0 || bucket.Size > 0):
		return &linodego.Error{
			Code: http.StatusBadRequest,
		}
	default:
		return &linodego.Error{
			Code: http.StatusNotFound,
		}
	}
```